docs/en/02_Developer_Guides/02_Controllers/05_Middlewares.md

```diff
@@ -118,6 +118,40 @@ SilverStripe\Control\Director:
       Controller: %$SpecialRouteMiddleware
 ```
 
+## Low-level application middleware
+
+Some use cases will require a middleware to run before the Silverstripe CMS has been fully bootstrapped (e.g.: Updating 
+the HTTPRequest before Silverstripe CMS routes it to a controller). This can be achieved by editing the Silverstripe 
+CMS entry point file.
+
+This file will be located in your own codebase at `public/index.php`, or directly in your project root at `index.php` 
+if your project doesn't use the public web root. Find the line that instantiate `HTTPApplication`. Call the 
+`addMiddleware` method on the `HTTPApplication` instance and pass it an instance of your middleware. This must be done
+before the request is handled.
+
+```php
+// Default application
+$kernel = new CoreKernel(BASE_PATH);
+$app = new HTTPApplication($kernel);
+
+$app->addMiddleware( new MyLowLevelApplicationMiddleware() );
+
+$response = $app->handle($request);
+$response->output();
+```
+
+Beware that by this point, many of the regular Silverstripe framework features you normally rely on won't be 
+available in your middleware or in `index.php` because they won't have been initialised yet.
```

Comment:
```suggestion
available in your middleware or in `index.php` because they won't have been initialised yet. (e.g.: ORM, Injector, services configured by Injector)
```